doc/direnv.txt

```diff
@@ -0,0 +1,167 @@
+
+Internally, this plugin remembers the sourced filenames and does not source it
+again even when `DirenvExport` is called in the same buffer.
+
+NOTE: Unlike the other features of Direnv, this “sourcing” is irreversible.
+You should NOT change settings globally in your `.vimrc.local` because it
+affects other windows/buffers.
+>
+    " in your .vimrc.local
+    " This sets the value only for the window you are in.
+    setlocal scrolloff=3
+    " Also set buffer-local variable for this use.
+    let b:ale_fixers = ['prettier', 'eslint']
+
+=============================================================================
+COMMANDS						      *direnv-commands*
+
+`:DirenvExport`					 *direnv-commands-direnvexport*
+
+Run the Direnv command and load the valid settings. This command will be
+executed automatically (and asynchronously) by |autocmd| events in default.
+
+NOTE: The kind of events this command will be called depends on your (Neo)
+Vim's version. See |direnv-caveats|.
+
+`:EditDirenvrc`					 *direnv-commands-editdirenvrc*
+
+Open the global setting file for the Direnv command. This searches files
+ordered below.
+
+* `$XDG_CONFIG_HOME/direnv/direnvrc`
+* `~/.config/direnv/direnvrc`
+* `~/.direnvrc`
+
+`:EditEnvrc`					    *direnv-commands-editenvrc*
+
+Open the detected `.envrc` if found or a new butter to edit `.envrc` on the
+current directory otherwise.
+
+=============================================================================
+CONFIGURATION						 *direnv-configuration*
+
+All configuration variables must be set before loading the plugin (typically
+in your `.vimrc` or `init.vim`).
+
+`g:direnv_auto`				      *direnv-configuration-direnvauto*
+
+It will not execute |direnv-commands-direnvexport| automatically if the value
+is `0`. Default: `1`.
+>
+    " Call DirenvExport on your demand
+    let g:direnv_auto = 0
+
+`g:direnv_edit_mode`			  *direnv-configuration-direnveditmode*
+
+Select the command to open buffers to edit. Default: `'edit'`.
+>
+    " split the window before editing files
+    let g:direnv_edit_mode = 'split'
+    " split vertically
+    let g:direnv_edit_mode = 'vsplit'
+
+`g:direnv_silent_load`			*direnv-configuration-direnvsilentload*
+
+Stop echoing output from Direnv command. Default: `0`.
+>
+    " Silently call the Direnv command
+    let g:direnv_silent_load = 1
+
+=============================================================================
+AUTOCMD EVENTS						 *direnv-autocmdevents*
+
+`DirenvLoaded`				    *direnv-autocmdevents-direnvloaded*
+
+This plugin calls this |User| event just after sourcing extra vimrc (See
+|direnv-extravimrc|).
+>
+    " Announce when loaded extra vimrc's
+    autocmd User DirenvLoaded :echo 'loaded extra vimrc'
+
+
+=============================================================================
+CAVEATS							       *direnv-caveats*
+
+Environmental variable keys~
+
+The Vimscript syntax seems to limit keys to alphanumeric characters. If any
```

Comment:
If direnv/direnv#901 gets merged in, this limitation will no longer apply since it uses `setenv(key,val)`. (But it's probably still worth having since that'll be a very new version of direnv)

Reply:
I see. It seems to need some logic to achieve Vim 8.0 compatibility which has no `setenv()`, I think.

Anyway, I will merge this PR with no change.